XenAdmin/TabPages/ManageUpdatesPage.cs

```diff
@@ -215,6 +220,219 @@ private bool FilterIsOn
+
+        private class UpdatePageDataGridViewRow : CollapsingPoolHostDataGridViewRow
+        {
+            private DataGridViewCell _poolIconCell;
+            private DataGridViewTextBoxCell _versionCell;
+            private DataGridViewImageCell _patchingStatusCell;
+            private DataGridViewTextBoxCell _statusCell;
+            private DataGridViewTextBoxCell _requiredUpdateCell;
+            private DataGridViewTextBoxCell _installedUpdateCell;
+
+            public UpdatePageDataGridViewRow(Pool pool)
+                : base(pool)
+            {
+                SetupCells();
+            }
+
+            public UpdatePageDataGridViewRow(Host host, bool hasPool)
+                : base(host, hasPool)
+            {
+                SetupCells();
+            }
+
+            public bool IsPoolOrStandaloneHost
+            {
+                get { return IsAPoolRow || (IsAHostRow && !HasPool); }
+            }
+
+            private void SetupCells()
+            {
+                if (IsPoolOrStandaloneHost)
+                    _poolIconCell = new DataGridViewImageCell();
+                else
+                    _poolIconCell = new DataGridViewTextBoxCell();
+
+                _nameCell = new DataGridViewTextAndImageCell();
+                _versionCell = new DataGridViewTextBoxCell();
+                _patchingStatusCell = new DataGridViewImageCell();
+                _statusCell = new DataGridViewTextBoxCell();
+                _requiredUpdateCell = new DataGridViewTextBoxCell();
+                _installedUpdateCell = new DataGridViewTextBoxCell();
+
+                Cells.AddRange(new[] { _poolIconCell, _nameCell, _versionCell, _patchingStatusCell, _statusCell, _requiredUpdateCell, _installedUpdateCell });
+
+                this.UpdateDetails();
+            }
+
+            // fill data into row
+            private void UpdateDetails()
+            {
+                Pool pool = Tag as Pool;
+
+                if (pool != null)
+                {
+                    Host master = pool.Connection.Resolve(pool.master);
+                    _poolIconCell.Value = Images.GetImage16For(pool);
+
+                    DataGridViewTextAndImageCell nc = _nameCell as DataGridViewTextAndImageCell;
+                    if (nc != null)
+                        nc.Image = null;
+
+                    _nameCell.Value = pool.Name;
+                    _versionCell.Value = master.ProductVersionTextShort;
+                    _requiredUpdateCell.Value = "";
+                    _installedUpdateCell.Value = "";
+
+                    var poolPatchingStatus = pool.Connection.Cache.Hosts.Any(h => Updates.RecommendedPatchesForHost(h).Count > 0);
+                    _patchingStatusCell.Value = poolPatchingStatus
+                        ? Properties.Resources._000_error_h32bit_16
+                        : Properties.Resources._000_Tick_h32bit_16;
+                    _statusCell.Value = poolPatchingStatus ? "Not Updated" : "Updated";

```

Comment:
Strings viewable on the UI belong to a resource file (normally we put then in XenModel/Messages.resx), so they can be localised for all supported languages (currently Chinese and Japanese).